### PR TITLE
Follow-Up Fixes for gRPC

### PR DESCRIPTION
**Pull request by kjin**
* Slight `wrap`-function doc change to reflect previous changes
* Comments summarizing start and end span points
* Check `agent.shouldTrace` to determine whether span should actually be started
* Renamed `func` object to `serverMethod` internally (less confusion)
* No longer dealing with `arguments` - `apply` calls changed to `call` with actual arguments
* Collapse consecutive `if` statements with identical conditions
* `switch` instead of `if...else if` statement for picking `wrap` function

### CODE REVIEW

File: src/hooks/userspace/hook-grpc.js [hunk 1]
```diff
@@ -144,8 +144,6 @@ function sendMetadataWrapper(rootContext) {
     return function sendMetadataTrace(responseMetadata) {
       if (rootContext) {
         rootContext.addLabel('metadata', JSON.stringify(responseMetadata.getMap()));
-      } else {
-        agent.logger.info('gRPC: No root context found in sendMetadata');
```
Comment (matthewloring) [inline src/hooks/userspace/hook-grpc.js]: Why is this removed?

**Comment by ofrobots** [comment]
Given that 'shouldTrace' sampling was a bugfix (the rest are mostly cosmetic, structure), would it be possible to add a test to make sure we do in fact have functional sampling that we don't regress in the future?

File: src/hooks/userspace/hook-grpc.js (hunk 1 above)
Comment (kjin) [inline src/hooks/userspace/hook-grpc.js]: `rootContext` cannot be null; its value here is always obtained directly from a call to `createRootSpanData`.

Comment (matthewloring) [inline src/hooks/userspace/hook-grpc.js]: If this is the case, the if statement above should be removed too and a brief comment added.

**Comment by kjin** [comment]
@ofrobots Added a test to show that `shouldTrace` is being called. Do you think I should also test this for the other hooks?

File: test/hooks/common.js [hunk 2]
```diff
@@ -136,6 +147,10 @@ function createChildSpan(cb, duration) {
   };
 }
 
+function getAndResetShouldTraceArguments() {
```
Comment (matthewloring) [inline test/hooks/common.js]: Is this used?

Comment (kjin) [inline test/hooks/common.js]: Nope, will remove
